android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -39,9 +40,22 @@ object WorkerBinder {
    * @param worker The class that wants to be informed when to start and stop doing work.
    * @return [WorkerUnbinder] to unbind [Worker&#39;s][Worker] lifecycle.
    */
+  @OptIn(DelicateCoroutinesApi::class)
   @JvmStatic
-  open fun bind(interactor: Interactor<*, *>, worker: Worker): WorkerUnbinder {
-    return bind(mapInteractorLifecycleToWorker(interactor.lifecycle()), worker)
+  public fun bind(interactor: Interactor<*, *>, worker: Worker): WorkerUnbinder {
+    val workerEvents = interactor.lifecycleFlow.map { it.asWorkerEvent() }
+    /*
+     * We need `Dispatchers.Unconfined` to react immediately to lifecycle flow emissions, and we need
+     * `CoroutineStart.Undispatched` to prevent coroutines launched in `onStart` with `Dispatchers.Unconfined`
+     * from forming an event loop instead of starting eagerly.
+     */
+    val job = GlobalScope.launch(RibDispatchers.Unconfined, start = CoroutineStart.UNDISPATCHED) {
```

Comment:
related PR for updated WorkerBinder (still pending test and updating the worker duration listener) https://github.com/uber/RIBs/pull/553/